mula/scheduler/schedulers/scheduler.py

```diff
@@ -417,6 +440,10 @@ def stop_threads(self) -> None:
 
         self.threads = []
 
+    def set_last_activity(self) -> None:
+        with self.lock:
+            self.last_activity = datetime.now(timezone.utc)
```

Comment:
This seems like a property to me rather than a setter method. Consider using `property.setter` if the intention is to apply a lock before assigning the variable